File: src/ros2_bridge/mappers/action_msgs/goal_status_array.rs
//! Typed mapper for `action_msgs/msg/GoalStatusArray`.

use crate::ros2_bridge::mapper::TypedTopicMapper;

pub(crate) fn goal_status_array_to_bus(
    msg: ros_env::action_msgs::msg::GoalStatusArray,
) -> crate::action_msgs::msg::v1::GoalStatusArray {
    crate::action_msgs::msg::v1::GoalStatusArray {
        status_list: msg
            .status_list
            .into_iter()
            .map(crate::ros2_bridge::mappers::action_msgs::goal_status::goal_status_to_bus)
            .collect(),
    }
}

pub(crate) fn goal_status_array_to_ros(
    bus: crate::action_msgs::msg::v1::GoalStatusArray,
) -> ros_env::action_msgs::msg::GoalStatusArray {
    ros_env::action_msgs::msg::GoalStatusArray {
        status_list: bus
            .status_list
            .into_iter()
            .map(crate::ros2_bridge::mappers::action_msgs::goal_status::goal_status_to_ros)
            .collect(),
    }
}

#[derive(Clone, Copy, Debug, Default)]
pub struct ActionMsgsGoalStatusArrayMapper;

impl TypedTopicMapper for ActionMsgsGoalStatusArrayMapper {
    type Ros = ros_env::action_msgs::msg::GoalStatusArray;
    type Bus = crate::action_msgs::msg::v1::GoalStatusArray;

    fn ros_to_bus(&self, msg: Self::Ros) -> crate::errors::Result<Self::Bus> {
        Ok(goal_status_array_to_bus(msg))
    }

    fn bus_to_ros(&self, msg: Self::Bus) -> crate::errors::Result<Self::Ros> {
        Ok(goal_status_array_to_ros(msg))
    }
}

File: bindings/python/robot_bus/ros2_bridge/mappers/action_msgs/goal_status_array.py
"""Generated mapper for `action_msgs/msg/GoalStatusArray`."""

from __future__ import annotations

from robot_bus.ros2_bridge.mappers import _convert
from robot_bus.ros2_bridge.mappers.action_msgs.goal_status import goal_status_to_bus, goal_status_to_ros

def goal_status_array_to_bus(msg):
    from robot_bus.action_msgs.msg.v1 import GoalStatusArray as BusMsg

    bus = BusMsg()
    bus.status_list.extend([goal_status_to_bus(x) for x in msg.status_list])
    return bus


def goal_status_array_to_ros(bus):
    from action_msgs.msg import GoalStatusArray as RosMsg

    out = RosMsg()
    out.status_list = [goal_status_to_ros(x) for x in bus.status_list]
    return out


class ActionMsgsGoalStatusArrayMapper:
    def type_name(self) -> str:
        return "action_msgs/msg/GoalStatusArray"

    def ros_msg_type(self):
        from action_msgs.msg import GoalStatusArray as RosMsg

        return RosMsg

    def ros_to_bus(self, msg) -> bytes:
        return goal_status_array_to_bus(msg).SerializeToString()

    def bus_to_ros(self, payload: bytes):
        from robot_bus.action_msgs.msg.v1 import GoalStatusArray as BusMsg

        bus = BusMsg()
        bus.ParseFromString(payload)
        return goal_status_array_to_ros(bus)
